lib/iris/cube.py

```diff
@@ -1984,6 +1984,12 @@ def coord(
             if name_or_coord is not None:
                 if not isinstance(name_or_coord, str):
                     _name = name_or_coord.name()
+                    emsg = (
+                        "Expected to find exactly 1 coordinate matching the "
+                        f"{_name!r} coordinate, but found none."
```

Comment:
```suggestion
                        "Expected to find exactly 1 coordinate matching the given"
                        f"{_name!r} coordinate's metadata, but found none."
```
Definitely an improvement but I wonder if we can be even more explicit.  I _think_ it's just the metadata that has to match, so you can have different points and bounds.